dnsimple/authentication.go

```diff
@@ -1,9 +1,24 @@
 package dnsimple
 
 import (
+	"context"
 	"net/http"
+
+	"golang.org/x/oauth2"
 )
 
+// BasicAuthClient returns a client that authenticates via HTTP Basic Auth with given username and password.
+func BasicAuthClient(_ context.Context, username, password string) *http.Client {
```

Comment:
Given this is an HTTP client, consider including that in the name of the function. `BasicAuthHTTPClient`